### ISSUE TITLE
False positive DirectMagicAttributeAccessViolation with __subclasses__

### ISSUE THREAD

**Issue by AlwxSin** [wemake-services/wemake-python-styleguide#1314]
# Bug report

## What's wrong
```python
class MyClass(object):
    """My class."""

subs = MyClass.__subclasses__()
```

This piece of code raises
`WPS609 Found direct magic attribute usage: __subclasses__`

## How is that should be
`__subclasess__` method should be excluded from the check because it's a class method and can be used outside of class.

Maybe there are other methods that should be excluded?

**Comment by sobolevn** [comment]
`__name__`?

**Comment by AlwxSin** [comment]
`__name__` doesn't raise violation 😁

But it should be excluded too, of course.

**Comment by mattany** [comment]
Hi, I have never contributed to an open-source project before but I would like to start here :) I have already forked the repository and cloned it. 
This is probably really dumb but I am having trouble understanding in what file this actual bug resides.
cheers

**Comment by sobolevn** [comment]
@mattany thanks a lot! And congrats on your first open-source fork 🎉 

I am going to help you!

1. We need to find the violation that @AlwxSin is talking about. It can be done even inside Github's search: https://github.com/wemake-services/wemake-python-styleguide/search?q=DirectMagicAttributeAccessViolation&unscoped_q=DirectMagicAttributeAccessViolation
2. Then we locate that it is used in for files: we are only interested in visitors. Read more about our architecture here: https://wemake-python-stylegui.de/en/latest/pages/api/index.html#overview
3. Then we find the exact place where this violation is raised in a visitor: https://github.com/wemake-services/wemake-python-styleguide/blob/58b96219a39390c35353aa9a29d483d5ed3f1cf7/wemake_python_styleguide/visitors/ast/attributes.py#L67

And let's start from here: we can create a whitelist of method names that can be called.
And do not raise violation if this method name is inside a whitelist.

Feel free to ask any questions!

**Comment by mattany** [comment]
Thank you! I noticed that in attributes.py there is already a whitelist like what you suggested:
```    _allowed_magic_attributes: ClassVar[FrozenSet[str]] = frozenset((
        '__class__',
        '__name__',
        '__qualname__',
        '__doc__',
    ))
```
It already includes ```__name__``` but doesn't include ```__subclasses__```. I can't think of any other magic methods/attributes that should be called directly in some contexts, but I believe that adding ```__subclasess__``` to the set above would fix this specific issue. If other false positives arise they can also be added there. 
Should I add it and send a pull request?

On the same subject, perhaps the excluded ones should be called from certain contexts and not from others? If there are contexts from which they too shouldn't be accessed directly, perhaps there is a way to discern between the 'good' and 'bad' calls? Just thought I would share in case we can make the exception even better.

**Comment by sobolevn** [comment]
Let's also add `__mro__` in the list and it's done!

**Comment by mattany** [comment]
Great. I issued a pull request. Hope I did it right   😄

**Comment by AlwxSin** [comment]
@sobolevn @mattany found the new one - `__version__`

**Comment by sobolevn** [comment]
@AlwxSin submit a PR then 🛷
